### PR TITLE
User/bulelani/cb 3163/compound tags add request services

### PR DESCRIPTION
**Pull request by bxaia**
@ofaletse or @ajoubertza Please review.

This PR includes adding the service handling creation of compound tags, and also the view side of the GUI. Attached is the screen shot of what userlog page would look like with compound tags populated.

screen shot 2018-08-28 at 18 05 03

### CODE REVIEW

File: app/services/userlog-service.js [hunk 1]
```diff
@@ -279,6 +282,49 @@
...
+        api.createCompoundTags = function (compound_tags, userlog_id) {
+            var promises = [];
+            for (var i = 0; i < compound_tags.length; i++) {
+                var defer = $q.defer();
+                var newCompoundTag = {
+                    value: compound_tags[i]
+                };
+                $http(createRequest('post', urlBase() + '/' + userlog_id + '/compound-tags/add', newCompoundTag)).then(
+                    function (result) {
+                        NotifyService.showSimpleToast("Created compound tag " + result.config.data.value + " id:" + result.data.id + ".");
```
[the remark this answers is not in the document]
Comment (bxaia) [reply app/services/userlog-service.js]: ok, will remove

File: app/userlogs/userlogs.html [hunk 2]
```diff
@@ -100,6 +103,9 @@
                     <span class="fa fa-paperclip"></span>)
                     </span>
                 </span>
+                <span style="min-width: 24px; margin-left: 8px; font-size: 12px">
+                    {{uselog.compound_tags.join(',')}}
```
[the remark this answers is not in the document]
Comment (bxaia) [reply app/userlogs/userlogs.html]: I'll try implement a similar counter as they did on the attachment and display a tag-icon with number of tags next to it

**Comment by ofaletse** [comment]
Please fix the horizontal scroll bar:
screen shot 2018-08-30 at 7 29 46 am

**Comment by ofaletse** [comment]
Look cool B, impressive.
Vertical scroll bar does not look ok to me. May I please ask that you fix the style of the _textarea_.

See what I am talking about:
screen shot 2018-09-04 at 7 56 38 am

The following screen shot looks better.
screen shot 2018-09-04 at 7 48 30 am

Consider changing code like this:
``` 
-                <textarea style="overflow-y: scroll; resize: none; width: 100%; min-height: 280px; max-height: 280px; height: 280px" ng-disabled="!editMode" md-select-on-focus rows="10" md-maxlength="10000" ng-model="content"></textarea>
+                <textarea style="overflow-y: scroll; resize: none; width: 100%; min-height: 190px; max-height: 190px; height: 190px" ng-disabled="!editMode" md-select-on-focus rows="10" md-maxlength="10000" ng-model="content"></textarea>

-                <textarea style="overflow-y: scroll; resize: none; min-height: 80px; max-height: 80px; height: 80px" ng-disabled="editMode" ng-list="&#10;" ng-trim="false" md-select-on-focus rows="2" md-maxlength="1000" ng-model="compound_tags"></textarea>
+                <textarea style="overflow-y: scroll; resize: none; min-height: 50px; max-height: 50px; height: 50px" ng-disabled="editMode" ng-list="&#10;" ng-trim="false" md-select-on-focus rows="2" md-maxlength="1000" ng-model="compound_tags"></textarea>
```

**Pull request merged**